security-framework/src/passwords.rs

```diff
@@ -26,9 +29,9 @@ use security_framework_sys::keychain_item::{
 
 /// Set a generic password for the given service and account.
 /// Creates or updates a keychain entry.
-pub fn set_generic_password(service: &str, account: &str, password: &[u8]) -> Result<()> {
+pub fn set_generic_password(service: &str, account: &str, password: &[u8], access_control_options: Option<AccessControlOptions>) -> Result<()> {
```

Comment:
You'll need to add this as a separate method, because `set_generic_password` is a stable API (promised to never change).

Perhaps it'd be a good time to introduce a builder-like pattern here, similar to what Apple does with the `query`. Create a `PasswordOptions` type that wraps the `Vec<(CFString, CFType)>` and sets query properties individually, and then a method like `set_password(PasswordOptions)`. This will be extensible for future properties that Apple may add.

Reply:
Sure I'll try to do something like this

Reply:
so i tried my best for this one, learning on the go, see my recent changes